examples/cross-browser.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2021 Google LLC.
+# Copyright (c) Microsoft Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script implements Puppeteer's `examples/cross-browser.js` scenario using WebDriver BiDi.
+# https://github.com/puppeteer/puppeteer/blob/4c3caaa3f99f0c31333a749ec50f56180507a374/examples/cross-browser.js
+
+import asyncio
+import json
+import os
+import websockets
+
+
+async def get_websocket():
+    port = os.getenv('PORT', 8080)
+    url = f'ws://localhost:{port}'
+    return await websockets.connect(url)
+
+
+async def send_JSON_command(command, websocket):
+    await websocket.send(json.dumps(command))
+
+
+async def read_JSON_message(websocket):
+    return json.loads(await websocket.recv())
+
+
+async def run_and_wait_command(command, websocket):
+    command_id = command["id"]
+    await send_JSON_command(command, websocket)
+
+    # Read messages until the sent command is done.
+    while True:
+        message = await read_JSON_message(websocket)
+        if "id" in message and message["id"] == command_id:
+            return message
+
+
+async def main():
+    # const browser = await puppeteer.launch();
+    # https://github.com/puppeteer/puppeteer/blob/4c3caaa3f99f0c31333a749ec50f56180507a374/examples/cross-browser.js#L29
+    websocket = await get_websocket()
+
+    # Not implemented:
+    # await browser.version());
+    # https://github.com/puppeteer/puppeteer/blob/4c3caaa3f99f0c31333a749ec50f56180507a374/examples/cross-browser.js#L32
+
+    # Puppeteer:
+    # const page = await browser.newPage();
+
+    # Part 1. Execute BiDi command and wait for result.
+    # ... await browser.newPage();
+    # https://github.com/puppeteer/puppeteer/blob/4c3caaa3f99f0c31333a749ec50f56180507a374/examples/cross-browser.js#L31
+    command_result = await run_and_wait_command({
+        "id": 1000,
```

Comment:
Can you hide the ID inside of `run_command`? The code shouldn't have to pick an ID I think.

Reply:
I don't want to hide the ID logic in a helper, as reading the protocol from the example would be harder.
https://github.com/GoogleChromeLabs/chromium-bidi/pull/68#discussion_r757911067